src/lib/db.js

```diff
@@ -342,9 +342,17 @@ export default class db {
     return new Promise((resolve) => {
       database.select('slug')
       .from('teams')
-      .orderBy('name', 'asc')
+      .innerJoin('teams_authors', 'teams.id', '=', 'teams_authors.team_id')
+      .orderBy('teams_authors.team_order', 'asc')
       .then((rows) => {
-        const result = _.map(rows, (row) => {return row.slug});
+        // The SQL is returning a copy of the team for every author. This
+        // solution manually filters the results and works but it's hacky.
+        // We don't use ES6 sets to preserve the order.
```

Comment:
Well as far as I can see in docs ES6 sets actually preserve order. Since numbers here will always be negligible, in the order of 3-5 teams at most always, so we can basically use any method we want I guess, but I guess since sets preserve order they're probably implemented as red-black trees with have O(logn) complexity so in general a better solution for this kind of thing would be a hash table I guess.